src/pages/dataElements/New.tsx

```diff
@@ -1,5 +1,183 @@
+    const attributeValues = customAttributes.map((attribute) => ({
+        attribute,
+        value: '',
+    }))
+
+    return {
+        name: '',
+        shortName: '',
+        code: '',
+        description: '',
+        url: '',
+        color: '',
+        icon: '',
+        fieldMask: '',
+        domainType: '',
+        formName: '',
+        valueType: '',
+        aggregationType: '',
+        categoryCombo: { id: '' },
+        optionSet: { id: '' },
+        commentOptionSet: { id: '' },
+        legendSets: [],
+        aggregationLevels: [],
+        attributeValues,
+    }
+}
+
+const ADD_NEW_DATA_ELEMENT_MUTATION = {
+    resource: 'dataElements',
+    type: 'create',
+    data: (de: object) => de,
+} as const
+
+function formatFormValues({ values }: { values: FormValues }) {
+    return {
+        aggregationLevels: values.aggregationLevels,
+        aggregationType: values.aggregationType,
+        attributeValues: values.attributeValues.filter(({ value }) => !!value),
+        categoryCombo: values.categoryCombo.id
+            ? values.categoryCombo
+            : undefined,
+        code: values.code,
+        commentOptionSet: values.commentOptionSet.id
+            ? values.commentOptionSet
+            : undefined,
+        description: values.description,
+        domainType: values.domainType,
+        fieldMask: values.fieldMask,
+        formName: values.formName,
+        legendSets: values.legendSets,
+        name: values.name,
+        optionSet: values.optionSet.id ? values.optionSet : undefined,
+        shortName: values.shortName,
+        style: {
+            color: values.style?.color,
+            icon: values.style?.icon,
+        },
+        url: values.url,
+        valueType: values.valueType,
+        zeroIsSignificant: values.zeroIsSignificant,
+    }
+}
+
+// @TODO(DataElements/new): values dynamic or static?
 export const Component = () => {
-    return <div>This is a form for creating a new Data Element</div>
+    const dataEngine = useDataEngine()
+
+    const navigate = useNavigate()
+    const customAttributesQuery = useCustomAttributesQuery()
+
+    const loading = customAttributesQuery.loading
+    const error = customAttributesQuery.error
+
+    if (error && !loading) {
+        // @TODO(Edit): Implement error screen
+        return `Error: ${error.toString()}`
+    }
+
+    if (loading) {
+        // @TODO(Edit): Implement loading screen
+        return 'Loading...'
+    }
+
+    const initialValues = computeInitialValues(customAttributesQuery.data)
+
+    async function onSubmit(values: FormValues) {
+        const payload = formatFormValues({ values })
+
+        try {
+            // We want the promise so we know when submitting is done. The promise
+            // returned by the mutation function of useDataMutation will never
+            // resolve
+            await dataEngine.mutate(ADD_NEW_DATA_ELEMENT_MUTATION, {
```

Comment:
Could potentially use useMutation directly?

Reply:
No, the `mutate` function of the useMutation hook never resolves...

Reply:
Haha, I wonder why that is intentional... But I meant using `useMutation` from `react-query` instead.

Reply:
We're not allowed to do that though, right?

Reply:
... I do it in my other PR because there's no alternative. Of course we can use it when it's hindering us.

Reply:
Ah, I see. In this case I guess it's not really needed